docs/users/sync-data.mdx

```diff
@@ -29,43 +21,53 @@ To enable webhooks, go to the [Webhooks](https://dashboard.clerk.com/last-active
 
 You'll be presented with a form where you can specify the URL of your backend endpoint. This is the URL where Clerk will send the webhook events. You can also specify the events you want to receive. For example, if you only want to receive events related to users, you can select the **user** option.
 
+<Callout>
+If you are developing on your localhost, you will need to expose your endpoint to the internet to work with webhooks. See the [Testing the webhook](#testing-the-webhook) section for information.
+</Callout>
+
 ![Add endpoint form](/docs/images/users/guides/sync-backend/webhook-form.png)
 
 Once you click the **Create** button, you'll be presented with your webhook endpoint dashboard. Here you can see the URL of your endpoint and the events you selected and you can also test your endpoint.
 
-### Setup your backend endpoint
+### Add your Signing Secret to your `.env.local` file
+
+
+To retrieve your Webhook Signing Secret, click on the [Webhooks](https://dashboard.clerk.com/last-active?path=webhooks) page in the side nav of the [Clerk Dashboard](https://dashboard.clerk.com). 
+<Images
+  width={2000}
+  height={1125}
+  src="/images/users/guides/sync-backend/sign-secret.png"
+  alt="The Webhooks page in the Clerk Dashboard. There is a red arrow pointing to where the Signing Secret is located."
+/>
+You will need to set this value as an environment variable in your project. This guide uses `WEBHOOK_SECRET` as the key. However, you can set the key to whatever you like; just be sure to update the code examples.
+```env filename=".env.local"
+WEBHOOK_SECRET=your_signing_secret
+```
+
+
+### Understanding the webhook payload
 
-Now that you have your webhook endpoint set up, you need to set up your backend endpoint to receive the events. The Clerk webhook events are sent as HTTP POST requests with a JSON body. The body of the request contains the event type and the data related to the event.
+The Clerk webhook events are sent as HTTP `POST` requests with a JSON body. All messages contain:
+- `data` - an object that holds information for the event's payload.
+- `object` - this is always `event`
+- `type` - the type of webhook event. See [Supported webhook events](/docs/integrations/webhooks#supported-webhook-events) for a full list.
 
-Here's an example of a webhook payload for a `user.created` event:
+Below is an example of a webhook object with no payload:
 
 ```json
 {
   "data": {
-    "birthday": "",
-    "created_at": 1654012591514,
-    "email_addresses": [
-      {
-        "email_address": "james@clerk.com"
-      }
-    ]
+    // The event type specific payload will be here.
```

Comment:
Invalid JSON. We don't allow it (see STYLEGUIDE) because if folks don't remove the comments after pasting, it can bugger things up. Any other way to do this?

Reply:
I don't know the best place to land here. The concern about copy and pasting is definitely legitimate, but shouldn't apply here. The developer shouldn't be copying this -- its just a stand in. 

I was trying to avoid using a specific example but instead trying to keep this agnostic. If we wanted a full example, I would say to go with a *.deleted one, such as the `user.deleted` sample:
```js
{
  "data": {
    "deleted": true,
    "id": "user_29wBMCtzATuFJut8jO2VNTVekS4",
    "object": "user"
  },
  "object": "event",
  "type": "user.deleted"
}
```

We definitely want something different that the current one which is valid JSON but is also a very incomplete (and maybe misleading) version of user.created.

```js
{
  "data": {
    "birthday": "",
    "created_at": 1654012591514,
    "email_addresses": [
      {
        "email_address": "james@clerk.com"
      }
    ]
  },
  "object": "event",
  "type": "user.created"
}
```

The 'Test the webhook' expands on test the webhook, but will also show all of the sample JSON objects. https://docs-preview-450.clerkpreview.com/docs/users/sync-data#test-the-webhook
We have more information on the payload at https://clerk.com/docs/integrations/webhooks#payload-structure

Reply:
Pity we can't do something like...
```json
{
  "data": {
    "birthday": "",
    "created_at": 1654012591514,
    "email_addresses": [
      {
        "email_address": "james@clerk.com"
      }
    ]
},
   [event_payload](https://clerk.com/docs/integrations/webhooks#payload-structure)
},
```

I'm not going to block this because it hasn't become a problem before, but let's start thinking about better ways to document JSON objects